### ISSUE TITLE
Incorrect "default value" error on certain subclassed dataclasses

### ISSUE THREAD

**Issue by drew** [microsoft/pyright#745]
**Describe the bug**
Subclassed dataclasses incorrectly show the following type error when the parent dataclass contains a `field` with `init=False`.

> Data class fields without default value cannot appear after data fields with default values

**To Reproduce**
```python
from dataclasses import dataclass, field


@dataclass
class Parent:
    prop_one: str = field(init=False)

    def __post_init__(self):
        self.prop_one = "test"


@dataclass
class Child(Parent):
    prop_two: str  # <- pyright shows a "default value" type error here


test = Child(prop_two="test")

assert test.prop_one == "test"
assert test.prop_two == "test"

```
**Expected behavior**
`prop_two` on the `Child` class should not display a type error since `prop_one` does not contain a default value.

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/385/85162548-6a42aa80-b216-11ea-9389-f0b8f331ffb3.png)

**VS Code extension or command-line**
* VSCode v1.46.1
* Pyright v1.1.43
* Python v.3.7.7

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.1.45, which I just published.
